File: exercise1/task4.js
const fs = require('fs');

function computeMeanCSV(filename, columnNumber) {
    const data = fs.readFileSync(filename, 'utf8').split('\n');
    let sum = 0;
    let count = 0;

    for (let i = 0; i < data.length; i++) {
        const row = data[i].split(',');
        const value = parseFloat(row[columnNumber]);

        if (!isNaN(value)) {
            sum += value;
            count++;
        }
    }

    if (count > 0) {
        return sum / count;
    } else {
        return null;
    }
}

const filename = 'csv_task4.csv'; // Replace with your CSV file name
const columnNumber = 0; // Replace with the column number (0-indexed)

const mean = computeMeanCSV(filename, columnNumber);
if (mean !== null) {
    console.log(`The mean of column ${columnNumber} is: ${mean}`);
} else {
    console.log('No data or invalid column number provided.');
}

File: exercise1/task4.py
import sys
import csv
import statistics

def computeMeanCSV(filename, column_number):
    data = []
    with open(filename, newline='') as csvfile:
        reader = csv.reader(csvfile)
        for row in reader:
            try:
                data.append(float(row[column_number]))
            except (ValueError, IndexError):
                pass
    
    if data:
        mean = statistics.mean(data)
        return mean
    else:
        return None


if __name__ == "__main__":
    if len(sys.argv) != 3:
        print("Usage: python csv.py [FILE] [COLUMN_NUMBER]")
        sys.exit(1)
    
    filename = sys.argv[1]
    column_number = int(sys.argv[2])
    
    mean = computeMeanCSV(filename, column_number)
    if mean is not None:
        print(f"The mean of column {column_number} is: {mean}")
    else:
        print("No data or invalid column number provided.")
